laurent22/joplin
# Android: Fixes #10681: Fix Dropbox sync for certain device languages

# Summary

This pull request should fix #10681 by patching `rn-fetch-blob`. Previously, `rn-fetch-blob` used the user-set locale to lower-case strings, **including all headers sent with HTTP requests**. This caused issues with the Turkish language, where the lower case form of `I` **is not** `i`.

**Notes**:
- This pull request patches `rn-fetch-blob`, rather than migrates to a maintained alternative. See https://github.com/laurent22/joplin/issues/10681#issuecomment-2214459100 for details.
- This roughly applies [this commit](https://github.com/RonRadtke/react-native-blob-util/commit/46876cb9c10a111631f4a6c497867af16a0d1e82) from the `react-native-blob-util` fork.

# Testing plan

On Android 13:
1. Set the device language to Turkish (Türkçe).
2. Re-authenticate to Dropbox.
3. Sync.
4. Verify that sync completes successfully.
5. Add a drawing to a note.
6. Sync.
7. Verify that sync completes successfully.
